tests/test_config.py

```diff
@@ -69,3 +70,43 @@ def test_BUGWARRIORRC_empty(self):
         os.environ['BUGWARRIORRC'] = ''
         rc = self.create('.config/bugwarrior/bugwarriorrc')
         self.assertEquals(config.get_config_path(), rc)
+
+
+class TestGetDataPath(ConfigTest):
+
+    def setUp(self):
+        super(TestGetDataPath, self).setUp()
+        self.config = ConfigParser.RawConfigParser()
+        self.config.add_section('general')
+
+    def test_TASKDATA(self):
+        """
+        TASKDATA should be respected, even when taskrc's data.location is set.
+        """
+        datapath = os.environ['TASKDATA'] = os.path.join(self.tempdir, 'data')
+
+        self.assertEqual(
+            datapath, config.get_data_path(self.config, 'general'))
+
+    def test_taskrc_datalocation(self):
+        """
+        When TASKDATA is not set, data.location in taskrc should be respected.
+        """
+        os.environ['TASKDATA'] = ''
+
+        self.assertEqual(
+            self.lists_path, config.get_data_path(self.config, 'general'))
+
+    def test_unassigned(self):
+        """
+        When data path is not assigned, raise an exception.
+        """
+        # Empty taskrc.
+        with open(self.taskrc, 'w'):
+            pass
+
+        os.environ['TASKDATA'] = ''
+
+        with self.assertRaisesRegexp(RuntimeError,
+                                     'Unable to determine the data location.'):
+            config.get_data_path(self.config, 'general')
```

Comment:
I'm not sure about this last case: it seems that taskwarrior itself doesn't require you to set `data.location` and will use the default `~/.task` if not set:
```sh
~ ❯❯❯ TASKRC=$(mktemp) task _show | grep data.location
data.location=~/.task
```